recipes/dotnet_agent.rb

```diff
@@ -0,0 +1,102 @@
+agent = node['appdynamics']['dotnet_agent']
+controller = node['appdynamics']['controller']
+proxy = node['appdynamics']['http_proxy'] 
+bitness = node['kernel']['os_info']['os_architecture']
+install_directory = agent['install_dir']
+system_directory = node['kernel']['os_info']['windows_directory']
+
+agent_msi = ""
+##################################################################################################################################
+# The below paths can also be used instead of temp direcotry. But there will be msiexec exceptions if there is space in the path #
+# agent_msi_path = "#{Chef::Config[:file_cache_path]"																			 #
+# setup_config = "#{Chef::Config[:file_cache_path]}\\setup.xml"																	 #
+# install_log_file = "#{Chef::Config[:file_cache_path]}\\DotnetAgentInstall.log"												 #
+##################################################################################################################################
+agent_msi_path = "#{system_directory}\\Temp"
+setup_config = "#{agent_msi_path}\\setup.xml"
+install_log_file = "#{agent_msi_path}\\DotnetAgentInstall.log"
+
+
+# Check the bitness of the OS to determine the installer to download and run
+case bitness
+when /64/
+  agent_msi = "#{agent_msi_path}\\dotNetAgentSetup64.msi"
+  source_path = "#{agent['source']}/dotNetAgentSetup64.msi"
+when /32/
+  agent_msi = "#{agent_msi_path}\\dotNetAgentSetup.msi"
+  source_path = "#{agent['source']}/dotNetAgentSetup.msi"
+else
+  agent_msi = "Unsupported OS bitness"
+end
+
+# Download the msi file from source
+remote_file "#{agent_msi}" do
+  source source_path
+  checksum agent['checksum']
+end
+
+# Environment Validation
+# Comment if you're installing 4.1 -- Matt Jensen to confirm with Sanjay
+# Check whether the MSDTC service is up and running 
+# if it is not running, start the service
+powershell_script 'check_MSDTC' do
```

Comment:
i think the powershell_script blocks that are doing a service check could be changed to use the standard chef  resource 'service'. the chef provider has the logic built into it to only enable it if its not running

service "MSDTC" do
  action [ :enable, :start ]
end
service "Winmgmt" do
  action [ :enable, :start ]
end
service "COMSysApp" do
  action [ :enable, :start ]
end